bundles/org.openhab.binding.pushsafer/src/main/java/org/openhab/binding/pushsafer/internal/actions/PushsaferActions.java

```diff
@@ -0,0 +1,229 @@
+
+    private @NonNullByDefault({}) PushsaferAccountHandler accountHandler;
+
+    @RuleAction(label = "@text/sendPushsaferMessageActionLabel", description = "@text/sendPushsaferMessageActionDescription")
+    public @ActionOutput(name = "sent", label = "@text/sendPushsaferMessageActionOutputLabel", description = "@text/sendPushsaferMessageActionOutputDescription", type = "java.lang.Boolean") Boolean sendPushsaferMessage(
+            @ActionInput(name = "message", label = "@text/sendPushsaferMessageActionInputMessageLabel", description = "@text/sendPushsaferMessageActionInputMessageDescription", type = "java.lang.String", required = true) String message,
+            @ActionInput(name = "title", label = "@text/sendPushsaferMessageActionInputTitleLabel", description = "@text/sendPushsaferMessageActionInputTitleDescription", type = "java.lang.String", defaultValue = DEFAULT_TITLE) @Nullable String title) {
+        logger.trace("ThingAction 'sendPushsaferMessage' called with value(s): message='{}', title='{}'", message,
+                title);
+        return send(getDefaultPushsaferMessageBuilder(message), title);
+    }
+
+    public static Boolean sendPushsaferMessage(ThingActions actions, String message, @Nullable String title) {
+        return ((PushsaferActions) actions).sendPushsaferMessage(message, title);
+    }
+
+    @RuleAction(label = "@text/sendPushsaferURLMessageActionLabel", description = "@text/sendPushsaferURLMessageActionDescription")
+    public @ActionOutput(name = "sent", label = "@text/sendPushsaferMessageActionOutputLabel", description = "@text/sendPushsaferMessageActionOutputDescription", type = "java.lang.Boolean") Boolean sendPushsaferURLMessage(
+            @ActionInput(name = "message", label = "@text/sendPushsaferMessageActionInputMessageLabel", description = "@text/sendPushsaferMessageActionInputMessageDescription", type = "java.lang.String", required = true) String message,
+            @ActionInput(name = "title", label = "@text/sendPushsaferMessageActionInputTitleLabel", description = "@text/sendPushsaferMessageActionInputTitleDescription", type = "java.lang.String", defaultValue = DEFAULT_TITLE) @Nullable String title,
+            @ActionInput(name = "url", label = "@text/sendPushsaferMessageActionInputURLLabel", description = "@text/sendPushsaferMessageActionInputURLDescription", type = "java.lang.String", required = true) String url,
+            @ActionInput(name = "urlTitle", label = "@text/sendPushsaferMessageActionInputURLTitleLabel", description = "@text/sendPushsaferMessageActionInputURLTitleDescription", type = "java.lang.String") @Nullable String urlTitle) {
+        logger.trace(
+                "ThingAction 'sendPushsaferURLMessage' called with value(s): message='{}', url='{}', title='{}', urlTitle='{}'",
+                message, url, title, urlTitle);
+        if (url == null) {
+            throw new IllegalArgumentException("Skip sending message as 'url' is null.");
```

Comment:
Your binding shouldn't throw unchecked exceptions if the user input is wrong. You could return `false` and log it.

Reply:
> @appzer What's the state of this PR?

@fwolter Hi Fabian, it is possible that you can help me with the desired changes so that the plugin can go online as soon as possible. Java is not my programming language, so I need help. Thank you

Reply:
Sure! How can I help you?

Reply:
@fwolter i made some changes and ask for help by some conversations!

Reply:
@fwolter Hi Fabian,
no luck in the developer forum! I'am not able to make the changes you wish! How can we proceed now?
Is there a way to hand in the current code for further development / completion.

Kevin

Reply:
That's some kind of a tough situation. If I take you by hand, this repo's backlog will increase even more rapidly, because I'd need to shift priorities. You could increase the propability to receive answers in the forum if you express your question way more concrete.

Reply:
From my POV throwing exceptions here is okay as the `ScriptActionHandler` can deal with them and print the exception message for the user. This way the user knows what is wrong with his method call.

```
14:39:56.041 [ERROR] [internal.handler.ScriptActionHandler] - Script execution of rule with UID 'a5f393dd1a' failed: Skip sending message as 'message' is null.
```